### PR TITLE
Add .devcontainer

### PR DESCRIPTION
**Pull request by jlsjonas**
Adds support for [devcontainers](https://containers.dev/), this should make onboarding easier and should allow (especially simpler) contributions entirely online via Github Codespaces 🚀

### CODE REVIEW

File: .devcontainer/devcontainer.json [hunk 1]
```diff
@@ -0,0 +1,43 @@
+// For format details, see https://aka.ms/devcontainer.json. For config options, see the
+// README at: https://github.com/devcontainers/templates/tree/main/src/docker-existing-docker-compose
+{
+	"name": "Actual development",
+
+	// Update the 'dockerComposeFile' list if you have more compose files or use different names.
+	// The .devcontainer/docker-compose.yml file contains any overrides you need/want to make.
```
Comment (j-f1) [inline .devcontainer/devcontainer.json:7]: Can you remove the comments like these unless they’re explaining something that a person reading this would need to know?

File: .devcontainer/devcontainer.json [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+
+	// The optional 'workspaceFolder' property is the path VS Code should open by default when
+	// connected. This is typically a file mount in .devcontainer/docker-compose.yml
+	"workspaceFolder": "/workspaces/${localWorkspaceFolderBasename}",
+
+	// Features to add to the dev container. More info: https://containers.dev/features.
+	// "features": {},
+
+	// Use 'forwardPorts' to make a list of ports inside the container available locally.
+	// "forwardPorts": [],
```
Comment (j-f1) [inline .devcontainer/devcontainer.json:27]: Shouldn’t 3001 be forwarded? (There may also be a separate WebSocket port, not sure)

File: .devcontainer/docker-compose.yml [hunk 3]
```diff
@@ -0,0 +1,20 @@
...
+  # Update this to the name of the service you want to work with in your docker-compose.yml file
+  actual-development:
+    # Uncomment if you want to override the service's Dockerfile to one in the .devcontainer 
+    # folder. Note that the path of the Dockerfile and context is relative to the *primary* 
+    # docker-compose.yml file (the first in the devcontainer.json "dockerComposeFile"
+    # array). The sample below assumes your primary file is in the root of your project.
+    #
+    # build:
+    #   context: .
+    #   dockerfile: .devcontainer/Dockerfile
```
Comment (j-f1) [inline .devcontainer/docker-compose.yml:12]: Similar to the JSON: can you clean up the comments in this file too?

File: CONTRIBUTING.md [hunk 4]
```diff
@@ -39,8 +39,11 @@ If you would like to contribute you can fork this repository and create a branch
 
 There are two options for developing:
```
Comment (j-f1) [inline CONTRIBUTING.md:40]: :)
```suggestion
There are three options for developing:
```

File: .devcontainer/docker-compose.yml [hunk 5]
```diff
@@ -0,0 +1,20 @@
...
+
+    volumes:
+      # Update this to wherever you want VS Code to mount the folder of your project
+      - ..:/workspaces:cached
+
+    # Overrides default command so things don't shut down after the process ends.
+    command: /bin/sh -c "while sleep 1000; do :; done"
```
Comment (j-f1) [inline .devcontainer/docker-compose.yml:19]: Why is this necessary? Shouldn’t the server keep running?

File: .devcontainer/devcontainer.json (hunk 1 above)
Comment (jlsjonas) [reply .devcontainer/devcontainer.json:7]: I wasn't sure about leaving them in or not, as the maintainers might not have been aware of the options.
Will remove (besides the image one)

File: .devcontainer/devcontainer.json (hunk 2 above)
Comment (jlsjonas) [reply .devcontainer/devcontainer.json:27]: This is mainly intended for "already running" services, devcontainer will detect & auto-forward 3001 to an available port (and inform you via UI)

File: .devcontainer/docker-compose.yml (hunk 5 above)
Comment (jlsjonas) [reply .devcontainer/docker-compose.yml:19]: This is some (default) devcontainer magic as the _default_ command is not how you'll interact with the container.
`yarn install` happens as a `postCreateCommand` (to be more transparent to the user)

I considered also starting the server there, but that would be against best practices. (hence "starting it in your preferred way") -- potential room for improvement there

File: .devcontainer/devcontainer.json (hunk 1 above)
Comment (j-f1) [reply .devcontainer/devcontainer.json]: They were definitely useful to look at on the first review pass, thanks for keeping them in! But yes would prefer to keep the final files clean

**Review by j-f1 — APPROVED** [review]
LGTM, thanks!
